src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -56,11 +56,11 @@
      * Returns an iterable view of all {@link Slot}s (leaf nodes) in this
      * Inventory.
      *
-     * @param <T> expected inventory type, specified as generic to allow easy
+     * @param <T> expected slot type, specified as generic to allow easy
      *      pseudo-duck-typing
      * @return an iterable view of all Slots (leaf nodes) in this inventory
      */
-    <T extends Inventory> Iterable<T> slots();
+    <T extends Slot> Iterable<T> slots();
```

Comment:
Leaf nodes are *currently* guaranteed to be `Slot`s but I definitely don't want this encoded in the API, because the entire point of this is that it can support future changes in minecraft's inventory system. This is why pretty much all generic signatures are `<T extends Inventory>`, because the contract is designed around the core premise that you deal with `Inventory`s not any particular subinterface.